jax/_src/compilation_cache.py

```diff
@@ -84,6 +84,8 @@ def is_cache_used(backend: xla_client.Client) -> bool:
         _cache_used = True
       return _cache_used
 
+  return False
```

Comment:
I don't think this line is reachable, but I agree it's better than implicitly returning `None`.

Reply:
Wouldn't it be better to assert False instead?